File: src/main/java/de/blazemcworld/fireflow/command/LobbyCommand.java
package de.blazemcworld.fireflow.command;

import de.blazemcworld.fireflow.space.Lobby;
import de.blazemcworld.fireflow.util.Transfer;
import de.blazemcworld.fireflow.util.Translations;
import net.kyori.adventure.text.Component;
import net.kyori.adventure.text.format.NamedTextColor;
import net.minestom.server.command.builder.Command;
import net.minestom.server.entity.Player;

public class LobbyCommand extends Command {

    public LobbyCommand() {
        super("lobby", "spawn");

        setDefaultExecutor((sender, ctx) -> {
            if (sender instanceof Player player) {
                if (player.getInstance() == Lobby.instance) {
                    sender.sendMessage(Component.text(Translations.get("error.already.lobby")).color(NamedTextColor.RED));
                    return;
                }

                Transfer.move(player, Lobby.instance);
            } else {
                sender.sendMessage(Component.text(Translations.get("error.needs.player")).color(NamedTextColor.RED));
            }
        });
    }

}

File: src/main/kotlin/de/blazemcworld/fireflow/command/LobbyCommand.kt
package de.blazemcworld.fireflow.command

import de.blazemcworld.fireflow.Lobby
import net.minestom.server.command.builder.Command
import net.minestom.server.command.builder.CommandExecutor
import net.minestom.server.entity.Player

object LobbyCommand : Command("lobby", "spawn", "hub") {
    init {
        defaultExecutor = CommandExecutor exec@{ sender, _ ->
            if (sender !is Player) return@exec
            if (sender.instance == Lobby.instance) return@exec
            Lobby.playerJoin(sender)
        }
    }
}